tincr/io/design/tincr_checkpoints.tcl

```diff
@@ -127,10 +128,23 @@ proc ::tincr::write_placement_xdc {args} {
             puts $xdc "set_property BEL [get_property BEL $cell] \[get_cells \{[get_name $cell]\}\]"
             puts $xdc "set_property LOC [get_property LOC $cell] \[get_cells \{[get_name $cell]\}\]"
             puts $txt "BEL [get_property BEL $cell] [get_cells [get_name $cell]]"  
-            puts $txt "LOC [get_name $cell] [get_property LOC $cell] [get_property BEL $cell] [get_tile -of [get_sites -of $cell]]"
+
+	    # To match the xdlrc file generated from TINCR's "write_xdlrc", IOB sites that are bonded
+	    # have been updated to use the package pin name rather than the actual site name.  
+	    # The Vivado .xdc files have not been changed, only the txt files used for parsing a design into rapidSmith2
+            set sitename [get_property LOC $cell]
+
+            if { [regexp {.*IOB*} $sitename]} {
```

Comment:
I agree with Brad’s suggestion.  It does not rely on names to differentiate.



From: Brad White <notifications@github.com<mailto:notifications@github.com>>
Reply-To: bradselw/tincr <reply@reply.github.com<mailto:reply@reply.github.com>>
Date: Tuesday, February 9, 2016 at 1:20 AM
To: bradselw/tincr <tincr@noreply.github.com<mailto:tincr@noreply.github.com>>
Subject: Re: [tincr] Add code to provide proper IOB site pin names for bonded IOB's. Only (#15)


In tincr/io/design/tincr_checkpoints.tcl<https://github.com/bradselw/tincr/pull/15#discussion_r52277756>:

> @@ -127,10 +128,23 @@ proc ::tincr::write_placement_xdc {args} {
>              puts $xdc "set_property BEL [get_property BEL $cell] \[get_cells \{[get_name $cell]\}\]"
>              puts $xdc "set_property LOC [get_property LOC $cell] \[get_cells \{[get_name $cell]\}\]"
>              puts $txt "BEL [get_property BEL $cell] [get_cells [get_name $cell]]"
> -            puts $txt "LOC [get_name $cell] [get_property LOC $cell] [get_property BEL $cell] [get_tile -of [get_sites -of $cell]]"
> +
> +         # To match the xdlrc file generated from TINCR's "write_xdlrc", IOB sites that are bonded
> +         # have been updated to use the package pin name rather than the actual site name.
> +         # The Vivado .xdc files have not been changed, only the txt files used for parsing a design into rapidSmith2
> +            set sitename [get_property LOC $cell]
> +
> +            if { [regexp {.*IOB*} $sitename]} {


I don't think it is a good idea to key off of a substring ("IOB") in the name like this. There's no contract between a site's name and its type. Better to use the site's properties instead. Consider how write_xdlrc does it:

    foreach site $sites {
        set state "internal"
        set name [get_property NAME $site]
        if {[get_property IS_PAD $site]} {
            if {[get_property IS_BONDED $site]} {
                set state "bonded"
                set name [get_property NAME [get_package_pins -quiet -of_object $site]]
            } else {
                set state "unbonded"
            }
        }


—
Reply to this email directly or view it on GitHub<https://github.com/bradselw/tincr/pull/15/files#r52277756>.

________________________________
________________________________